src/CNRECS.DEF

```diff
@@ -669,6 +669,56 @@ RECORD PY4 "4th power poly sub" *SUBSTRUCT	// 4th power polynomial. for GT, whic
     *f *ARRAY 6 FLOAT k			// FLOAT k[6].  One extra member needed as terminator by cul.cpp array input logic.
 *END		// PY4
 //==================================================================================================================
+RECORD PERFORMANCEMAP "PERFORMANCEMAP" *RAT
+
+    *prefix pm_
+    *declare "RC pm_CkF();"
+    *declare "RC pm_SetupBtwxt( record* pParent, const char* tag, class Btwxt::RegularGridInterpolator* &pRgi, double& capRatRef, double& ratingSpeedF) const;"
+    *declare "RC pm_GetGridAxis( int iGX, PMGRIDAXIS*& pGXRet) const;"
+    *declare "RC pm_GetLookupData( int iLU, PMLOOKUPDATA*& pLURet) const;"
+    *declare "RC pm_GetLookupValue( int iLU, float& luValRet, int iGx1, int iGx2) const;"
+    *declare "RC PERFORMANCEMAP::pm_GXCheckAndMakeVector( int iGX, PMGRIDAXIS* &pGX, std::vector< double>& vGX, std::pair< int, int> sizeLimits) const;"
+    *declare "RC PERFORMANCEMAP::pm_LUCheckAndMakeVector( int iLU, PMLOOKUPDATA* &pLU, std::vector< double>& vLU, int expectedSize, double vMin=-DBL_MIN, double vMax=DBL_MAX) const;"
+
+    *i SI pm_dummy      // *declare cannot be last (rcdef restriction / shortcoming)
+
+*END    // PERFORMANCEMAP
+//==================================================================================================================
+RECORD PMGRIDAXIS "PMGRIDAXIS" *RAT
+    *prefix pmx_
+
+    *ovrcopy
+    *declare "RC pmx_CkF();"
+    *declare "PERFORMANCEMAP* pmx_GetPERFMAP() const;"
+    *declare "RC pmx_CheckAndMakeVector( std::vector< double>& vGX, std::pair< int, int> sizeLimits) const;"
+
+    *i CULSTR pmx_id        // axis identifier (e.g. "DBT")
+                            // cannot use PMGRIDAXIS.name
+                            //   multiple PMGRIDAXISs may use same pmx_id
```

Comment:
Given this is only used for error/warning messages, I don't really understand the need to have a separate name/id from the record name. If the message says "DBT", you still need to figure out which "DBT".

Reply:
Error messages also include object name.

Reply:
Why do the error messages need a name and an id when an axis can already be uniquely identified by its name?

Reply:
pmx_id is something the user can make up.  It is used in error messages and passed to btwxt as the gridaxis name.

We are going round and round on this.  The original performance map design had defined types with strict structural requirements.  You argued for a more free-form design that could be used with some generality.  I reimplemented that way.  I now think this was a mistake because it is all too easy to make errors that cannot be detected (except via crazy results).  We should leave this alone pending a more careful redesign.